working/declaring-constructors/feature-specification.md

```diff
@@ -278,11 +352,44 @@ the parameter. A declaring constructor can be `const` or not. It can't be
 redirecting. It also can't be `external` since an external constructor has no
 implicit initializer list or body where the fields could be initialized.*
 
+We also need to allow `var` before a simple formal parameter while also allowing
+a type annotation. (Today, `var` is allowed, but only in place of a type, like
+how variables are declared.) We redefine `simpleFormalParameter` to:
+
+```
+simpleFormalParameter ::=
+    'covariant'? ('final' | 'var')? type? identifier
+```
+
+*The `simpleFormalParameter` rule was previously defined in terms of the same
+rules used for variable declarations. That meant that the grammar for parameters
+allowed `late` and `const`. Those are then disallowed by the specification
+outside of the grammar. Here, we eliminate the need for that extra-grammatical
+restriction by defining a grammar specifically for simple formal parameters that
+only includes what they allow.*
+
+*We could allow `late` on field parameters and have that apply to the instance
+field (but not the parameter since parameters are always initialized). That
+could be useful in theory if there are other generative constructors that don't
+initialize the field. But to avoid confusion, we simply don't allow it. If a
+user wants a `late` instance field, they can always declare it outside of the
+declaring constructor.*
+
+It is a compile-time error for a `simpleFormalParameter` to have both `var`
+and a type outside of a declaring constructor.
+
+*This keeps the normal parameter list grammar consistent with other variable
+declarations. We could allow `var int x` as a parameter outside of a declaring
+constructor, but doing so would be confusing because it looks like a field
+parameter but isn't. Ideally, we would also disallow `final` on parameters
+outside of declaring constructors, but doing so is a breaking change.*
```

Comment:
Disallowing `final` would probably have to be combined with `final`-as-default, otherwise *some* people will be complaining (because we'd have to choose being mutable as the behavior for all parameters, lest we break a lot of code).
(Or they won't notice and won't complaine, because the same people who insist on `final` on all other local variables also consistently forget to put it on parameters. Maybe it's just because it's not linted.)